ddtrace/opentracer/tracer.py

```diff
@@ -85,6 +117,40 @@ def start_span(self, operation_name=None, child_of=None, references=None,
         """"""
         pass
 
+    def inject(self, span_context, format, carrier):
+        """Injects a span context into a carrier.
+
+        :param span_context: span context to inject.
+
+        :param format: format to encode the span context with.
+
+        :param carrier: the carrier of the encoded span context.
+        """
+        if not isinstance(carrier, dict):
```

Comment:
why we're expecting this to be a `dict`? it should work with any kind of object, assuming there is a proper `propagator` defined. No?

Reply:
True, the check should be performed in the propagator where the type is implied, the tracer shouldn't care about the type of the carrier. I'll move these checks to the propagator.

Reply:
Perfect!